File: app/src/main/java/com/example/delivery/room/DaoMenu.java
package com.example.delivery.room;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.delivery.data.Favorites;
import com.example.delivery.data.MyEntity;
import com.example.delivery.data.Param;

import java.util.List;

import io.reactivex.Maybe;
import io.reactivex.Observable;
import io.reactivex.Single;

@Dao
public interface DaoMenu {
    @Query("SELECT *FROM myentity WHERE id=:id")
    Maybe<MyEntity> getPosition(int id);


    @Query("SELECT * FROM myentity INNER JOIN favorites WHERE myentity.id=favorites.entityId ")
    Maybe<List<MyEntity>> getFavorites();

    @Query("SELECT * FROM favorites")
    Maybe<List<Favorites>> getlistFavorites();

    @Insert
    long saveFavoriteOne(Favorites entityId);

    @Delete
    int deleteFavoriteOne(Favorites favorites);

    @Query("DELETE FROM favorites WHERE entityId=:entityId")
    Single<Integer> deleteFavoriteOne(int entityId);

    @Query("SELECT * FROM favorites WHERE entityId=:entityId")
    Maybe<Favorites> cheakOne(int  entityId);
}

File: app/src/main/java/com/example/deliverykotlin/room/DaoMenu.kt
package com.example.deliverykotlin.room

import androidx.annotation.NonNull
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.example.deliverykotlin.data.Brand
import com.example.deliverykotlin.data.MyEntity
import com.example.deliverykotlin.data.Type

@Dao
interface DaoMenu {

    @Insert
    fun saveEntityList(@NonNull entities:List<MyEntity>): Array<Long>

    @Insert
    fun saveThirstParam(@NonNull params: List<Brand>): Array<Long>

    @Insert
    fun saveSecondParam(@NonNull params: List<Type>): Array<Long>

    @Query("SELECT * FROM myentity")
    fun loadEntityList():List<MyEntity>

    @Query("SELECT * FROM brand")
    fun loadThirstParamList():List<Brand>

    @Query("SELECT * FROM type")
    fun loadSecondParamList():List<Type>

}